Assets/Scripts/CopilotMechanics/MushroomFriendActive.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class MushroomFriendActive : CopilotActiveMechanic
+{
+    public GameObject HealingBubblePrefab;
+    private HealingBubble healingBubbleScript;
+    private GameObject healingBubble;
+    public int heal; //amount to heal player
+    public float healTime; //time to restore health
+    public bool healPlayer; //check that Player is allowed to heal
+
+
+    void Start()
+    {
+        healPlayer = false;
+    }
+
+    void Update()
+    {
+        //if Player can heal, activate Player active ability and show healing bubble
+        if (healPlayer)
+        {
+            if (InputManager.GetButtonDown(PlayerInput.PlayerButton.ActiveAbility))
+            {
+                if (healingBubble == null)
+                {
+                    healingBubble = Instantiate(HealingBubblePrefab, PlayerInfo.singleton.transform);
+                    healingBubbleScript = healingBubble.GetComponent<HealingBubble>();
+                    healingBubbleScript.SetSporeTimer(healTime);
+                }
+                else
+                {
+                    healingBubbleScript.SetSporeTimer(healTime);
+                }
+            }
+            if (healingBubbleScript != null && !healingBubbleScript.GetHealPlayer())
+            {
+                healPlayer = false;
+            }
+        }
+
+        //if Player health is 25% or lower, allow Player to heal
+        if (!healPlayer && PlayerInfo.singleton.health <= PlayerInfo.singleton.maxHealth * 0.25f)
```

Comment:
Try not to use float values like this cuz you never know when a designer wants to change it. Make an instance variable and use it instead of the 0.25f